core/modules/mysql/MySqlConfig.cc

```diff
@@ -55,23 +55,20 @@ MySqlConfig::MySqlConfig(std::string const& username,
                          std::string const& hostname,
                          unsigned int const port,
                          std::string const& socket,
-                         std::string const& dbName)
+                         std::string const& dbName,
+                         bool const checkValid)
     : username(username), password(password), hostname(hostname), port(port),
       socket(socket), dbName(dbName) {
 
-    if ((hostname.empty() or port == 0) and socket.empty()) {
-        LOGS(getLogger(), LOG_LVL_FATAL, "MySQL hostname:port and socket both undefined");
-        throw std::invalid_argument("MySQL hostname:port and socket both undefined" );
+    if (checkValid) {
+        checkValidity();
```

Comment:
Well, if you want to check that configuration is good I have nothing against that (if you can do it in a reasonable way). But this class has nothing to do with configuration, there is no assumption anywhere that instances of this class are going to be created from user configuration only. I think it is too low-level to make any assumptions at all.